### PR TITLE
Do not check for multicast support when making DNSSD default advertise decisions.

### PR DESCRIPTION
**Pull request by andy31415**
#### Problem
mDNS uses multicast, however the flag IFF_MULTICAST does not seem to be always set even if supporting it.
In particular, `tun0` interfaces used in podman (maybe docker too) will support multicast but will not advertise as such.

#### Change overview
Remove the multicast check completely. Our listen options are generally just "report failure" and will cope with partial errors anyway. We seem to want to try harder to work by default instead of being more conservative.

#### Testing
Was able to locally run a test for CHIP-Repl in an environment where IPv6 multicast was not working (due to podman setup which is harder to track down).

### CODE REVIEW

File: examples/minimal-mdns/AllInterfaceListener.h [hunk 1]
```diff
@@ -88,14 +88,24 @@ class AllInterfaces : public mdns::Minimal::ListenIterator
         }
     }
 
+    template <typename T>
+    inline bool MulticastOk(T & iterator)
```
Comment (andy31415) [inline examples/minimal-mdns/AllInterfaceListener.h:92]: This is a lot of code duplication, however the logic of skiping seems generally duplicated, I did not try to centralize as well in this PR.

**Review by woody-apple — APPROVED** [review]

**Comment by andy31415** [comment]
@woody-apple @bzbarsky-apple @tcarmelveilleux 

Raising here discussions regarding this: mDNS is "multicast dns" so using broadcast for detecting multicast support is wrong. But it does work and I see LWIP using that flag in its implementation.

So this leads me to believe that IFF_MULTICAST is broken somehow, but debugging this sounds painful. Also the point was made of 'what is an IP stack without multicast'? 

Is this change useful or just makes the problem deeper? Should I just not use `SupportsMulticast` at all in minmdns code? maybe that is simpler.

**Comment by tcarmelveilleux** [comment]
> Is this change useful or just makes the problem deeper? Should I just not use `SupportsMulticast` at all in minmdns code? maybe that is simpler.

I would argue that if use need to use minmdns, you should only reply on interfaces that support multicast, but doing a check that an interface supports multicast is moot, since if that interface doesn't support multicast and you need mDNS, Matter stack won't work anyway.

Overall the root cause is we *try to advertise on all interfaces indiscriminately* as opposed to binding to N interfaces (where N is usually actually only 1 for real environments) where we are wishing to run Matter. If multiple interfaces are valid, then you are possibly multi-homing, and if they are separate VLANs, then the nodes on one may not see another, so you are already being "split mind" as a node. The topologies supported by spec are clear: one main VLAN with N stub networks across border routers reaching that VLAN.

To summarize:

- We need to get to "you specify the interface over which you want to run Matter" on Linux. Auto-doing-it on all, at best will be needless, at worse will cause weird interop issues
- For now, disregard `MulticastSupported` and ignore interfaces where you fail to advertise

**Comment by andy31415** [comment]
Trying to figure out the issue, it seems that tun interfaces in general will not advertise knowing multicast because they depend on the other end for this.

So in my case (and probably others) the other end does allow multicast and it works, just our check for detecting it is not sufficient.

**Review by tcarmelveilleux — APPROVED** [review]

**Comment by bzbarsky-apple** [comment]
> We need to get to "you specify the interface over which you want to run Matter" on Linux. Auto-doing-it on all, at best will be needless, at worse will cause weird interop issues

It already causes issues on darwin, fwiw, which is why chip-all-clusters-app now supports an "advertise only on this interface" argument....
